packages/vm/test/api/t8ntool/t8ntool.spec.ts

```diff
@@ -34,11 +47,209 @@ const args: T8NOptions = {
+    const common = new Common({
+      hardfork: Hardfork.Prague,
+      eips: [663, 3540, 3670, 4200, 4750, 5450, 6206, 7069, 7480, 7620, 7692, 7698],
+      chain: Mainnet,
+    })
+    const sm = new MerkleStateManager({ common })
+    const vm = await createVM({ common, stateManager: sm })
+    const code = hexToBytes('0xef000101000402000100030400010000800001305000ef')
+
+    const pk = randomBytes(32)
+    const caller = createAddressFromPrivateKey(pk) // caller address
+    const contractAddress = createAddressFromPrivateKey(randomBytes(32)) // contract address
+
+    await vm.stateManager.putCode(contractAddress, code)
+    await vm.stateManager.putAccount(caller, new Account(BigInt(0), BigInt(0x11111111)))
+
+    const tx = await createTx({
+      gasLimit: BigInt(0xffff),
+      gasPrice: 0x7,
+      to: contractAddress,
+    }).sign(pk)
+    const trace: string[] = []
+    vm.evm.events!.on('step', (step) => {
+      trace.push(JSON.stringify(stepTraceJSON(step, true)))
+    })
+    vm.events!.on('afterTx', async (event) => {
+      trace.push(JSON.stringify(await summaryTraceJSON(event, vm)))
+    })
+    const result = await runTx(vm, {
+      tx,
+      skipBalance: true,
+      skipNonce: true,
+      skipBlockGasLimitValidation: true,
+    })
+    assert.equal(result.execResult.executionGasUsed, BigInt(4))
+    assert.equal(trace.length, 4)
+  })
+  it('should produce a trace with storage activated', async () => {
+    const bytecode = hexToBytes('0x604260005560206000f3') // PUSH1 42 PUSH1 00 MSTORE PUSH1 20 PUSH1 00 RETURN
+    const common = new Common({ chain: Mainnet, hardfork: Hardfork.Cancun })
+    const sm = new MerkleStateManager({ common })
+    const vm = await createVM({ common, stateManager: sm })
+    const contractAddress = createAddressFromPrivateKey(randomBytes(32))
+    await vm.stateManager.putAccount(contractAddress)
+    await vm.stateManager.putCode(contractAddress, bytecode)
+    const trace: string[] = []
+    vm.evm.events!.on('step', (step) => {
+      trace.push(JSON.stringify(stepTraceJSON(step, false, true)))
+    })
+    vm.events!.on('afterTx', async (event) => {
+      trace.push(JSON.stringify(await summaryTraceJSON(event, vm)))
+    })
+    const tx = await createTx({
+      to: contractAddress,
+      gasLimit: 0xffffffff,
+      gasPrice: 0xf,
+    }).sign(randomBytes(32))
+
+    await runTx(vm, { tx, skipBalance: true, skipBlockGasLimitValidation: true, skipNonce: true })
+    assert.equal(trace.length, 7, 'trace length is 7')
+
+    // First step trace should have empty storage
+    const traceStepWithoutStorage = JSON.parse(trace[0])
+    expect(traceStepWithoutStorage.storage.length).toBe(0)
+
+    // Last step trace should have storage with actual value
+    const traceStepWithStorage = JSON.parse(trace[5])
+    assert.exists(traceStepWithStorage.storage)
+    expect(traceStepWithStorage.storage).toMatchObject([['0x0', '0x42']])
```

Comment:
Should the storage not also include what address this storage is written to?

Reply:
I'll ask.  The spec is underspecified here.  It just says `storage -- Key-Value -- Array of all stored values`